core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -83,21 +83,6 @@ public static <T> CloseableIterable<T> markDeleted(
         });
   }
 
-  /**
-   * Returns the remaining rows (the ones that are not deleted), while counting the deleted ones.
-   *
-   * @param rows the rows to process
-   * @param isDeleted a predicate that determines if a row is deleted
-   * @return the processed rows
-   * @deprecated Will be removed in 1.2.0, use {@link Deletes#filterDeleted(CloseableIterable,
-   *     Predicate, DeleteCounter)}.
-   */
-  @Deprecated
-  public static <T> CloseableIterable<T> filterDeleted(
```

Comment:
I guess this is okay? It doesn't seem bad to have a version that doesn't require a counter.

Reply:
This is functionality that was adjusted by #4588. We then re-introduced + deprecated the method in #6146 to not have breaking API changes. Also there's currently no place in the code that would use this, so it seems better to remove it?